Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -2559,7 +2576,273 @@ def get_indicator_dbot_object(indicator):
+    elif command_status != 'Completed' or action_status == 'InProgress':
+        # schedule next poll
+        polling_args = {
+            'interval_in_seconds': interval_in_secs,
+            'polling': True,
+            **args
+        }
+
+        scheduled_command = ScheduledCommand(
+            command=cmd,
+            next_run_in_seconds=interval_in_secs,
+            args=polling_args,
+            timeout_in_seconds=600)
+
+        command_result = CommandResults(scheduled_command=scheduled_command)
+        return command_result
+
+    # action was completed
+    else:
+        return post_polling_process(client, command_result.outputs)
+
+
+def get_live_response_result_command(client, args):
+    machine_action_id = args['machine_action_id']
+    command_index = arg_to_number(args['command_index'])
+    res = client.get_live_response_result(machine_action_id, command_index)
+    file_link = res['value']
+
+    # download link, create file result
+    f_data = client.download_file(file_link)
+    try:
+        outputs = f_data.json()
+    except Exception:
+        outputs = {'value': file_link}
+
+    demisto.results(fileResult('Response Result', f_data.content))
+
+    return CommandResults(
+        outputs_prefix='MicrosoftATP.LiveResponseResult',
+        outputs=outputs,
+        readable_output=f'file_link: {file_link}'
+    )
+
+
+def get_machine_action_command(client, args):
+    id = args['machine_action_id']
+    res = client.get_machine_action_by_id(id)
+
+    return CommandResults(
+        outputs_prefix='MicrosoftATP.MachineAction',
+        outputs=res
+    )
+
+
+# -------------- Run Script ---------------
+
+def run_live_response_script_with_polling(client, args):
+    return run_polling_command(client, args, 'microsoft-atp-live-response-run-script', run_live_response_script_action,
+                               get_machine_action_command, get_successfull_action_results_as_info)
+
+
+def run_live_response_script_action(client, args):
+    machine_id = args['machine_id']
+    scriptName = args['scriptName']
+    comment = args['comment']
+    arguments = args.get('arguments')
+    params = [{
+        "key": "ScriptName",
+        "value": scriptName
+    }]
+    if arguments:
+        params.append(
+            {
+                "key": "Args",
+                "value": arguments
+            }
+        )
+    request_body = {
+        "Commands": [
+            {
+                "type": "RunScript",
+                "params": params
+            },
+        ],
+        "Comment": comment
+    }
+
+    # create action:
+    res = client.create_action(machine_id, request_body)
+    return CommandResults(
+        outputs={'action_id': res['id']}
+    )
+
+
+def get_successfull_action_results_as_info(client, res):
+    machine_action_id = res['id']
+    file_link = client.get_live_response_result(machine_action_id, 0)['value']
```

Comment:
is it possible to not get `value` back?

Reply:
Only if an exception has occurred. In this case I have to raise it anyway in order to show the actual error (action not completed, action already exist, etc)